spylib/token.py

```diff
@@ -282,3 +300,45 @@ def load(cls, store_name: str, associated_user: str):
         This method handles loading the token. By default this does nothing,
         therefore the developer should override this.
         """
+
+
+async def get_token(
+    domain: str,
+    code: str,
+    api_key: str,
+    api_secret_key: str,
+    CustomToken: Union[Type[OfflineTokenABC], Type[OnlineTokenABC]],
```

Comment:
Because we don't define `save` and `load` we cannot simply treat the input or return values as type `Token`. Technically we could just define abstract methods in `Token` but this would break the `SOLID` principles you were talking about